core/src/main/java/org/apache/iceberg/ManifestFilterManager.java

```diff
@@ -370,14 +407,7 @@ private boolean canContainDeletedFiles(ManifestFile manifest) {
       canContainDroppedFiles = false;
     }
 
-    boolean canContainDropBySeq =
-        manifest.content() == ManifestContent.DELETES
-            && manifest.minSequenceNumber() < minSequenceNumber;
-
-    return canContainExpressionDeletes
-        || canContainDroppedPartitions
-        || canContainDroppedFiles
-        || canContainDropBySeq;
+    return canContainExpressionDeletes || canContainDroppedPartitions || canContainDroppedFiles;
```

Comment:
Moved the canContainDropBy seq check first because it's possible that a manifest is not referenced but still needs to be rewritten in the case there are aged out deletes.